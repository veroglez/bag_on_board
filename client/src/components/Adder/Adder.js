import React from 'react';
import { useStateValue } from 'common/context/contextHooks';
import { Counter, Button, Input } from '..';
import style from './Adder.module.scss';

const onSubmit = (event, { cards, form }, dispatch) => {
  event.preventDefault();
  dispatch({
    type: 'update',
    payload: {
      cards: [...cards, { ...form }],
    },
  });
};

const Adder = () => {
  const [state, dispatch] = useStateValue();
  const { l10n } = state;

  return (
    <form className={style.container}>
      <Counter />
      <Input />
      <Button id="ButtonForm" text={l10n.ADD} onClick={event => onSubmit(event, state, dispatch)} />
    </form>
  );
};

export default Adder;
