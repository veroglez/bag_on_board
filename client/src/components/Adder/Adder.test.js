
import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as Context from 'common/context/contextHooks';
import Adder from './Adder';

Enzyme.configure({ adapter: new Adapter() });

const STATE = {
  l10n: { ADD: 'add' },
  cards: [
    { name: 'Vero', bags: 1 },
  ],
  form: { name: 'Veronica', bags: 2 },
};

const HOC = () => (
  <Context.StateContext.Provider value={[STATE]}>
    <Adder />
  </Context.StateContext.Provider>
);

describe('<Adder>', () => {
  it('render', () => {
    const tree = renderer.create(<HOC />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
