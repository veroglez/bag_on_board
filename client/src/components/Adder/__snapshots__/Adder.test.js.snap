// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Adder> render 1`] = `
<form
  className="container"
>
  <div
    className="container"
  >
    <button
      className="button"
      disabled={false}
      id="buttonMinus"
      onClick={[Function]}
      type="submit"
    >
      <span />
    </button>
    <p
      className="text"
    >
      2
    </p>
    <button
      className="button"
      disabled={false}
      id="buttonPlus"
      onClick={[Function]}
      type="submit"
    >
      <span />
    </button>
  </div>
  <input
    onChange={[Function]}
    type="text"
  />
  <button
    className="button"
    disabled={false}
    id="ButtonForm"
    onClick={[Function]}
    type="submit"
  >
    <span>
      add
    </span>
  </button>
</form>
`;
